toolkit/tools/scheduler/schedulerutils/preparerequest.go

```diff
@@ -76,6 +93,17 @@ func buildNodesToRequests(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildSta
 		}
 
 		defaultNode := buildNodes[0]
+
+		// Check if we already queued up this build node for building.
+		if buildState.IsSRPMBuildActive(defaultNode.SRPMFileName()) || buildState.IsNodeProcessed(defaultNode) {
+			err = fmt.Errorf("unexpected duplicate build for (%s)", defaultNode.SRPMFileName())
+			// Temporarily ignore the error, this state is unexpected but not fatal. Error return will be
+			// restored later once the underlying cause of this error is fixed.
+			logger.Log.Errorf(err.Error())
```

Comment:
Since this workaround should prevent builds from failing due to duplicate builds, I'd switch to warning right now. Otherwise, when a build fails for a different reason, this will likely be a red herring confusing the dev doing the debugging.

Reply:
```suggestion
			logger.Log.Warnf(err.Error())
```